### PR TITLE
Performance improvement attempt

### PR DESCRIPTION
**Pull request by crackcomm**
There is a slight difference in initialization, previously outputs were initialized as zeros, now are initialized in one tensor with biases as ones. It might be easily fixed but there is no degradation of performance and I believe it's backward compatible.

#1

### CODE REVIEW

File: neat/phenotype/feed_forward.py [hunk 1]
```diff
@@ -1,65 +1,56 @@
...
-        # Set input values
-        for u in input_units:
-            outputs[u.ref_node.id] = x[0][u.ref_node.id]
-
-        # Set bias value
-        for u in bias_units:
-            outputs[u.ref_node.id] = torch.ones((1, 0)).to(device)[0]
+        inputs = torch.index_select(x, 1, self.input_indices)
+        extras = torch.ones((1, self.extras_len), dtype=dtype, device=device)
+        output_tensor = torch.cat((inputs, extras), 1)
```
Comment (ddehueck) [inline neat/phenotype/feed_forward.py:34]: Seems to cause this error:

```
Traceback (most recent call last):
  File "xor_run.py", line 19, in <module>
    solution, generation = neat.run()
  File "/neat/population.py", line 26, in run
    genome.fitness = max(0, self.Config.fitness_fn(genome))
  File "/neat/experiments/xor/config.py", line 58, in fitness_fn
    pred = phenotype(input)
  File /anaconda3/envs/pytorch-neat/lib/python3.5/site-packages/torch/nn/modules/module.py", line 493, in __call__
    result = self.forward(*input, **kwargs)
  File "/neat/phenotype/feed_forward.py", line 34, in forward
    output_tensor = torch.cat((inputs, extras), 1)
RuntimeError: Expected object of scalar type Float but got scalar type Double for sequence element 1 in sequence argument at position #1 'tensors'
```

**Review by ddehueck — COMMENTED** [review]
Did you record any metrics on performance improvement?

**Comment by ddehueck** [comment]
This PR should resolve #3

**Comment by crackcomm** [comment]
Performance was same but looked more continuous and stable on the chart.

File: neat/phenotype/feed_forward.py (hunk 1 above)
Comment (ddehueck) [reply neat/phenotype/feed_forward.py:34]: @crackcomm I assume you don't get this error. What version of torch and python are you running?

Comment (crackcomm) [reply neat/phenotype/feed_forward.py:34]: python 3.7 and torch 1.0.1

it seems input tensor is missing `dtype = torch.float64`

**Pull request closed without merging**